### PR TITLE
Change wheel cache path

### PR DESCRIPTION
**Pull request by tdejager**
Change the path of the wheel cache.

### CODE REVIEW

File: crates/rattler_installs_packages/src/wheel_builder/mod.rs [hunk 1]
```diff
@@ -150,7 +150,9 @@ impl<'db, 'i> WheelBuilder<'db, 'i> {
             env_markers,
             wheel_tags,
             resolve_options,
-            locally_built_wheels: WheelCache::new(wheel_cache_dir.to_path_buf()),
+            locally_built_wheels: WheelCache::new(
+                wheel_cache_dir.to_path_buf().join("wheel_cache"),
```
Comment (wolfv) [inline crates/rattler_installs_packages/src/wheel_builder/mod.rs:154]: Should this be more descriptive?
One thing I dislike is that the user doesn't have full control over this location now.

Comment (wolfv) [reply crates/rattler_installs_packages/src/wheel_builder/mod.rs:154]: Yeah, maybe? Would that work?

Comment (tdejager) [reply crates/rattler_installs_packages/src/wheel_builder/mod.rs]: Okay @wolfv, changed it, does this better reflect what you want?

I've made it move by value as the cache is not really an object but more or less just wraps the location that the cache is in, this is how cacache works.

What I think is the downside is that it is easy to forget to add the `join('wheels')`. We could also add the `WheelCache` or path to the `PackageDb`. WDYT?

File: crates/rattler_installs_packages/src/artifacts/sdist.rs [hunk 2]
```diff
@@ -264,7 +264,7 @@ mod tests {
             &env_markers,
             None,
             &resolve_options,
-            package_db.1.path(),
+            WheelCache::new(package_db.1.path().join("wheels")),
```
Comment (wolfv) [inline crates/rattler_installs_packages/src/artifacts/sdist.rs:267]: Should this be wheels or should it mention something like "local_wheels"? I think we might have to do some pruning on these cached wheels more often vs. ones that come from      `pypi` (e.g. if the system changes).

Comment (tdejager) [reply crates/rattler_installs_packages/src/artifacts/sdist.rs]: Yeah so w.r.t to artifacts we are actually storing the http request/response pairs that serve as both a check for staleness and such, but also to get the `artifact` be at an `sdist` or `wheel` from. The wheel cache for locally built wheels is seperate, so these will not intersect on disk.

**Comment by tdejager** [comment]
I've made it so that the PackageDb now has the `WheelCache` including the other caches that it has.

**Review by wolfv — APPROVED** [review]

**Pull request merged**